src/interfaces/python/factory_python.i

```diff
@@ -0,0 +1,184 @@
+%{
+#include <shogun/util/visitors/InterfaceTypeVisitor.h>
+
+// we can use:
+// SWIG_TypeQueryModule or SWIG_MangledTypeQueryModule or SWIG_Python_TypeQuery
+// to query the swig_type in run time and be less implementation dependent maybe?
+// but this is faster and should not cause problems
+#define SHOGUN_GET_SWIG_TYPE(name)                                             \
+	SWIGTYPE_p_std__shared_ptrT_shogun__##name##_t
+
+namespace shogun
+{
+	class ShogunInterfaceToPyObject : public InterfaceTypeVisitor
+	{
+	public:
+		virtual void on(std::shared_ptr<SGObject>* v)
```

Comment:
minor thing, but now we are trying to always use `override` instead of overwriting virtual functions
```suggestion
		void on(std::shared_ptr<SGObject>* v) override
```

Reply:
override FTW